packages/connect-popup/src/index.tsx

```diff
@@ -44,91 +51,7 @@ const escapeHtml = (payload: any) => {
     }
 };
 
-// handle messages from window.opener and iframe
-const handleMessage = (
```

Comment:
I am spliting this handler to multiple smaller handlers. Some of the handlers are one time, meaning they may be unregistered after they did their job